Common/Sources/Extensions/String.swift

```diff
@@ -79,6 +79,30 @@ public extension String {
         }
     }
 
+    /// Format an amount to a currency format with currency code
+    ///
+    /// - Parameter currencyCode: the currency code
+    /// - Returns: a formatted String amount with currency code
+    func formatWithActualDecimal(with currencyCode: String?) -> String {
+        if !self.isEmpty {
+            let currencyFormatter = NumberFormatter()
+            let locale = Locale(identifier: Locale.preferredLanguages[0])
+            let localizedDecimalSeparator: Character = locale.decimalSeparator?.first ?? "."
+            let components = self.split(separator: localizedDecimalSeparator)
+            let fractionDigits = components.count <= 1 ? 2 : components[1].count
+            currencyFormatter.maximumFractionDigits = fractionDigits
+            currencyFormatter.minimumFractionDigits = fractionDigits
+            currencyFormatter.minimumIntegerDigits = 1
+            currencyFormatter.numberStyle = .currency
+            currencyFormatter.currencyCode = currencyCode
+            currencyFormatter.currencySymbol = ""
+            let formattedAmountInNumber = currencyFormatter.number(from: self)
+            return currencyFormatter.string(for: formattedAmountInNumber) ?? ""
+        } else {
+            return ""
```

Comment:
You could return a blank string on the first line of the method with a ternary operator

Reply:
or use `guard` statement

Reply:
It has been written as existing function.